### ISSUE TITLE
"TypeScript: Restart TS Server" no longer displayed in command palette

### ISSUE THREAD

**Issue by paro-paro** [microsoft/vscode#206654]
<img width="608" alt="Screenshot 2024-03-02 at 02 07 37" src="https://github.com/microsoft/vscode/assets/43136874/c7875a6e-40af-4e0c-af50-d89d2f581a59">

I am on latest macos version 1.87

Version: 1.87.0 (Universal)
Commit: 019f4d1419fbc8219a181fab7892ebccf7ee29a2
Date: 2024-02-27T23:42:56.944Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.1.0

**Comment by mjbvz** [comment]
Do you have a js/ts file open?

**Comment by jcready** [comment]
I'm running into this as well and yes, I'm looking at a TS file. If I "Developer: Reload Window" then all the normal TS commands reappear.
![image](https://github.com/microsoft/vscode/assets/358876/6f783ff1-08d9-49d2-bd5b-4b51604d12cf)

After reloading the window
![image](https://github.com/microsoft/vscode/assets/358876/d9d888bb-d3ea-4d75-9514-f0181f6e89f4)

**Comment by michaelschufi** [comment]
For me, it's happening frequently enough to be interrupting my workflows negatively (reloading the window makes you lose e.g. undo/redo commands). But it's still hard to reproduce on demand. 

So far, I've seen it happen when switching branches, (re-)installing packages or overwhelming the TS server with some multi-file refactoring (e.g. export renaming).
Such cases have historically made the TS server crash sometimes. But up until VS Code 1.86.0, I could just restart it using the command and it was not a big deal.

**Comment by paro-paro** [comment]
Hi @mjbvz 

Sorry, I opened the issue out of a frustrating coding session and did not add any reproduction/context at all. Really sorry for that 🙏 

I, indeed, had a TS file opened at the time of this issue being reported. It was showing a very peculiar behavior, hard to reproduce, sometimes showing and sometimes not and I did not take the time to properly debug it. 

I believe I was having trouble due to a major bump (on Friday night) of Volar (Vue Language Server) which sometimes conflicts with the built-in Typescript Language. 

In any case, everything is being working fine for me this week (after a few re-installs and re-boots).
